test/js/style/style_layer.test.js

```diff
@@ -93,10 +77,9 @@ test('StyleLayer#setPaintProperty', function(t) {
             }
         });
 
-        layer.resolvePaint({});
         layer.setPaintProperty('background-color', null);
 
-        t.deepEqual(layer.getPaintProperty('background-color'), [0, 0, 0, 1]);
+        t.equal(layer.getPaintProperty('background-color'), undefined);
```

Comment:
:warning: This is a breaking change; however the old behavior was "wrong"

# Old Behavior

```js
var layer1 = new StyleLayer({ ..., paint: { } });
layer.getPaintProperty('background-color') // undefined

var layer2 = new StyleLayer({ ..., paint: { 'background-color': 'red' } });
layer.setPaintProperty('background-color', null)
layer.getPaintProperty('background-color') // [0, 0, 0, 1]
```

# New Behavior

```js
var layer1 = new StyleLayer({ ..., paint: { } });
layer.getPaintProperty('background-color') // undefined

var layer2 = new StyleLayer({ ..., paint: { 'background-color': 'red' } });
layer.setPaintProperty('background-color', null)
layer.getPaintProperty('background-color') // undefined
```